config/application.rb

```diff
@@ -72,20 +72,265 @@ class Application < Rails::Application
+        :address => 'mail',
+        :port => 25,
+        :domain => nil,
+        :enable_starttls_auto => true,
+        :password => nil,
+        :authentication => nil,
+      }
+    else
+      s = s.smtp_settings
+      tmp = s.address ? s.address.to_s : ''
+      ex.address = tmp.length > 0 ? tmp : 'mail'
+      i = s.port ? s.port.to_i : 0
+      ex.port = i > 0 ? i : 25
+      tmp = s.domain ? s.domain.to_s : ''
+      ex.domain = tmp.length > 0 ? tmp : nil
+      tmp = s.enable_starttls_auto ? s.enable_starttls_auto.to_s : ''
+      ex.enable_starttls_auto = tmp == 'true'
+      ex.password = s.password.to_s if s.password
+      ex.authentication = s.authentication.to_s if s.authentication
+    end
+    tmp = (s && s.delivery_method) ? s.delivery_method.to_s : ''
+    ex.delivery_method = (tmp.in? ['sendmail', 'smtp', 'file', 'test']) \
+      ? tmp.intern
+      : 'sendmail'.intern
+
+    tmp = Settings.name ? Settings.name : ''
+    config.name = tmp.length < 1 ? 'MyOntohub' : tmp
+	Settings.name = config.name
+
+    tmp = Settings.asset_host ? Settings.asset_host.to_s : ''
+    config.action_controller.asset_host = tmp if tmp.length > 0
+
+    config.display_head_commit = Settings.display_head_commit \
+      ? Settings.display_head_commit.to_s == 'true'
+      : Rails.env != 'production'
+    Settings.display_head_commit = config.display_head_commit
+
+	Settings.display_symbols_tab =
+      Settings.display_symbols_tab && Settings.display_symbols_tab.to_s=='true'
+
+    Settings.format_selection =
+      Settings.format_selection && Settings.format_selection.to_s == 'true'
+
+    if ! Settings.footer
+      # only if nil!
+      Settings.footer = [
+        OpenStruct.new('text'=>'Imprint', 'href'=>'http://about.ontohub.org/'),
+        OpenStruct.new('text'=>'Source Code',
+          'href'=>'https://github.com/ontohub/ontohub')
+      ]
+    end
+
+    i = Settings.access_token && Settings.access_token.expiration_minutes \
+      ? Settings.access_token.expiration_minutes.to_i
+      : 0
+    Settings.access_token = i > 0 ? i : 360
+
+    # for devise config, only
+    i = Settings.allow_unconfirmed_access_for_days \
+      ? Settings.allow_unconfirmed_access_for_days.to_i
+      : -1
+    Settings.allow_unconfirmed_access_for_days = i < 0 ? 3 : i
+
+	# deprecated keys
+    rs = Settings.fallback_commit_email \
+      ? Settings.fallback_commit_email.to_s
+      : 'websvc'
+    r = Settings.fallback_commit_user \
+      ? Settings.fallback_commit_user
+      : "websvc@#{config.fqdn}"
+    s = Settings.external_repository_name \
+      ? Settings.external_repository_name.to_s
+      : 'External'
+
+    if ! Settings.git
+      Settings.git = OpenStruct.new('data_dir'=>'/data/git',
+        'verify_url'=>Settings.hostname, 'default_branch'=>'master',
+        'push_priority'=>
+          OpenStruct.new('commits'=>1, 'changed_files_per_commit'=>5),
+        'fallbacks'=>OpenStruct.new('user'=>rs, 'email'=>r, 'repo_name'=>s)
```

Comment:
Align the elements of a hash literal if they span more than one line.<br>Surrounding space missing for operator '=>'.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.